orttraining/orttraining/test/training_api/trainer/trainer.cc

```diff
@@ -152,38 +154,62 @@ void InitSyntheticDataLoader(
   }
 }
 
-void RunTraining(const TestRunnerParameters& params) {
-  const auto& api = Ort::GetApi();
-
-  CheckpointState state;
-  // TODO: update using public API's calling pattern, e.g. api.LoadCheckpoint().
-  EnforceCheck(LoadCheckpoint(params.checkpoint_to_load_path, state).IsOK(), "Failed to load checkpoint");
-
-  OrtSessionOptions* session_options = nullptr;
-  EnforceCheck(api.CreateSessionOptions(&session_options) == nullptr, "Failed to create session options.");
+#define ORT_RETURN_ON_ERROR(expr)                            \
+  do {                                                       \
+    OrtStatus* onnx_status = (expr);                         \
+    if (onnx_status != NULL) {                               \
+      auto code = g_ort_api->GetErrorCode(onnx_status);          \
+      const char* msg = g_ort_api->GetErrorMessage(onnx_status); \
+      g_ort_api->ReleaseStatus(onnx_status);                     \
+      printf("Run failed with error code :%d\n", code);          \
+      printf("Error message :%s\n", msg); \
+      return -1; \
+    }                                                        \
+  } while (0);
+
+int RunTraining(const TestRunnerParameters& params) {
+  g_ort_api = OrtGetApiBase()->GetApi(ORT_API_VERSION);
+
+  // Create Env
+  OrtEnv* env;
+  OrtThreadingOptions* threading_options = nullptr;
+  ORT_RETURN_ON_ERROR(g_ort_api->CreateThreadingOptions(&threading_options));
```

Comment:
just for my understanding, what are these threading options, and why do we need them in our test runner?

Reply:
Threading Options are required when using global threadpool... I was seeing failures when trying to set it up so ended up removing some code and added a todo to enable global threadpool